lib/autoproj/cli/daemon.rb

```diff
@@ -0,0 +1,112 @@
+require 'autoproj/cli/inspection_tool'
+require 'autoproj/github_watcher'
+require 'tmpdir'
+
+module Autoproj
+    module CLI
+        # Actual implementation of the functionality for the `autoproj daemon` subcommand
+        #
+        # Autoproj internally splits the CLI definition (Thor subclass) and the
+        # underlying functionality of each CLI subcommand. `autoproj-daemon` follows the
+        # same pattern, and registers its subcommand in {MainDaemon} while implementing
+        # the functionality in this class
+        class Daemon < InspectionTool
+            attr_reader :watcher
+            def initialize(*args)
+                super
+                ws.config.load
+                @update_failed = false
+            end
+
+            # Loads all package definitions from the installation manifest
+            #
+            # @return [Array<Autoproj::InstallationManifest::Package,
+            #     Autoproj::InstallationManifest::PackageSet>] package array
+            def resolve_packages
+                installation_manifest = Autoproj::InstallationManifest
+                    .from_workspace_root(ws.root_dir)
+                installation_manifest.each_package.to_a +
+                    installation_manifest.each_package_set.to_a
+            end
+
+            # Adds a VCS definition to the watch list of the internal
+            # GithubWatcher instance
+            #
+            # @param [Hash] vcs Hash representation of an Autoproj::VCSDefinition
+            # @return [Boolean] whether the given hash was valid or not
+            def watch_vcs_definition(vcs)
+                return false unless vcs[:type] == 'git'
+                return false unless vcs[:url] =~
+                    /(?:(?:https?:\/\/|git@).*)github\.com/i
+                return false unless vcs[:url] =~
```

Comment:
I'm trying to move away from `${N}`. Isn't rubocop setup this way ?

~~~
return false unless (m = URL_TO_REPO_RX.match(vcs[:url]))

owner = m[1]
name = m[2]
~~~

and setting the regexp as a constant in the class (which allows to test it separately)